2019-2020/Curs01/C/06-ecuatie_grad2.c

```diff
@@ -0,0 +1,56 @@
+/* Programarea Calculatoarelor CA, 2019-2020
+
+   Programul rezolva o ecuatie de gradul 2 : a*x^2 + b*x + c = 0  */
+
+#include <stdio.h>
+#include <math.h>  // pentru functia sqrt(radical) si fabs(modul)
+
+int main() {
+    double a, b, c, x1, x2, eps = 0.0001;
+    /* eps este eroarea pe care o folosim cand comparam numere reale (float, double) 
+       vom considera ca x == y, daca |x - y| < eps (ex: eps = 0.0001)  */
+    scanf("%lf%lf%lf", &a, &b, &c);
+
+    /* Pentru ca ecuatia sa fie de gradul II consideram ca a este diferit de 0
+       Exercitiu suplimentar: tratati cazul in care a = 0 (ecuatie de gradul I);  */
+
+    double delta = b * b - 4 * a * c;
+
+    if (delta < 0) {
+        printf("Ecuatia nu are solutii reale\n");
+    } else if (fabs(delta - 0) < eps) {
```

Comment:
I think this condition should be checked first, since (due to errors) delta might be also a very small negative double which should be considered 0.